### DISCUSSION TITLE
Unable to run percy snapshot locally and in CI

### DISCUSSION THREAD

**Discussion by AxonC** [percy/cli#768] [Support]
I'm experiencing an issue trying to run Percy locally and in a CI environment.

If I use the --debug flag on the command it works fine (as in does all the steps including downloading Selenium etc.). The stack trace appears to suggest based upon the source code this is an error with communicating with the Percy API?

I've refreshed the token to eliminate that - same error with both tokens set in the environment variable.

Command run: `npx percy snapshot . --verbose` (verbose for extra logging).

Version: `1.0.0-beta.72`
Node: `v14.17.1`

Azure DevOps Pipeline Log:
(Identical to local stack trace)
```
[percy:config] Found config file: .percy.yml (0ms)
[percy:config] Using config:
{
  version: 2,
  snapshot: {
    enableJavaScript: true,
    minHeight: 1024,
    widths: [
      375,
      1280
    ]
  },
  static: {
    include: '**/*.snapshot.html'
  }
} (17ms)
[percy:client] Creating a new build... (229ms)
[percy:cli:snapshot] Error: sha: must be filled; message: must be filled; committed-at: must be filled; parents: must be an array
    at handleFinished (/home/vsts/work/1/s/node_modules/@percy/client/dist/request.js:266:17)
    at IncomingMessage.<anonymous> (/home/vsts/work/1/s/node_modules/@percy/client/dist/request.js:282:27)
    at IncomingMessage.emit (node:events:402:35)
    at endReadableNT (node:internal/streams/readable:1343:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) (739ms)
```

**Comment by AxonC** [comment]
I realise this is the wrong place to raise this. I'll use the support channel. Thanks.

**Comment by Robdel12** [comment]
Hey @AxonC! This is a good place for it -- those support requests just make their way back to me and @wwilsman in the end. 

It looks like the build is failing to be created, is there a proper git SHA in the CI env? Can you check the values of these env vars and make sure it's a proper full git SHA? https://github.com/percy/cli/blob/master/packages/env/src/environment.js#L96


Edit; actually I see you say this happens locally too. That is pretty interesting 🤨 What's the git situation like for this project?

**Reply by AxonC** [reply]
Thanks for the response, @Robdel12 ! Nothing too special - it's just a single repository with Git tags. 

This has first been noticed as an issue on our canary builds (last successful on 17th December 2021) and we haven't made any changes to this repository since then. Nothing else has changed (famous last words I know).

The SHA on the branch I'm currently working looks like: `87a5bcd56343e008736122e561c1eaf1bb0ed39f`. Equally, changing branches and checking out a commit via a detached head close to the aforementioned date doesn't have any effect on this either. 

Edit: worth nothing too I've invoked the CLI using just `percy snapshot .` and also using `npx percy snapshot .` as specified above.

**Reply by Robdel12** [reply]
hmm! Do you have your [azure repo connected to your Percy project? ](https://docs.percy.io/docs/azure-devops)

**Reply by AxonC** [reply]
The repo appears in the 'integrations' tab and is selected appropriately - assuming this is set up correctly on this evidence. The only thing we haven't yet done is enable the visual approval steps in the branch policies.

**Reply by Robdel12** [reply]
My hunch is there's something going wrong with the integration. Azure Devops has shipped breaking changes on us multiple times now, so I'm going to assume that's happening again. If you'd like, could you remove the integration from Percy and see if a build is created? I'm pretty sure it would work once the integration is removed.

In any case, I've given a heads up to the eng team to take a look at what's going on with azure.

**Reply by AxonC** [reply]
I can do that tomorrow (almost 7pm here) and give some feedback. The only note I'd add is that it doesn't explain the local failure (I know you advise always running in the CI environment as per docs, but this was my last resort to test).

Thanks for the feedback so far - has been a bit of a head scratcher.

**Reply by Robdel12** [reply]
When a build is created the API will try and talk to the source code integration (azure here), so that's why it'll fail everywhere. Hopefully we'll have an answer for what's going on by your morning! 😃

**Reply by AxonC** [reply]
I did some digging in our main release pipeline and it appears to have worked as recently as 26th Jan, not 17th December 21 as I previously mentioned as happening in our canary pipeline.

**Reply by AxonC** [reply]
Hi @Robdel12 I don't suppose you've managed to make any progress on a resolution here?

**Reply by Robdel12** [reply]
@AxonC This one is out of my hands and in the engineering teams hands -- @j000shDotCom should have some time this week to investigate and see what's going on with Azure.

**Reply by AxonC** [reply]
@Robdel12 Hi - not sure if any changes were made, but it now is running locally again!

However, in the pipelines, it is now producing a different error.

```
[percy:client] Creating a new build... (1ms)
[percy:core] Failed to create build (1255ms)
[percy:core] Error: 500 Internal Server Error
    at handleFinished (file:///home/vsts/work/1/s/node_modules/@percy/client/dist/utils.js:173:17)
    at IncomingMessage.<anonymous> (file:///home/vsts/work/1/s/node_modules/@percy/client/dist/utils.js:191:27)
    at IncomingMessage.emit (node:events:538:35)
    at endReadableNT (node:internal/streams/readable:1345:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) (1ms)
[percy:core] Uploading 0 snapshots...
[percy:core:browser] Closing browser (20ms)
[percy:core:browser] Browser closed (41ms)
[percy:core] Build not created (1ms)
```
